### PR TITLE
completed the popup window and refactored the related product widgets by using data provided by the server

### PR DESCRIPTION
**Pull request by DevonL1010**

### CODE REVIEW

File: src/components/related/ProductCardList.jsx [hunk 1]
```diff
@@ -3,35 +3,40 @@ import axios from 'axios';
...
+  //   .then(response => setOverviewProduct(response.data))
+  // }, []);
 
   // console.log(relatedID)
 
   return (
     <div className={styles.productCardList}>
-      {relatedID.map((result, index) => {
+      {Object.values(related).map((result, index) => {
+        // console.log(result)
```
Comment (merosenlund) [inline src/components/related/ProductCardList.jsx:36]: You have a lot of commented out code throughout these files and it makes it a little bit hard to follow. If you have a section of code to the point that it is working it would be nice to get rid of all console.logs. 

As far as the other commented out code goes I am kind of the same way and don't like getting rid of code in case I need it again in the future. I have been trying to force myself to not allow this to happen though because it makes it harder to follow. I would encourage you to look through your commented out code and remove any that you are pretty sure you won't need to use anymore.

Another option for code that you won't need to use again for the current project but you want to remember how you did it you could copy and paste to your engineering journal.

**Review by merosenlund — CHANGES_REQUESTED** [review]
Hey Devon, overall this looks pretty good. There are some questions I have though and it looks like this has quite a bit of commented out code that could be cleaned up. 

I think in general we want to have the code we are merging with main be as clean as possible because that is our 'production' code. However if some of the commented code you think you are going to need for future features feel free and keep it in.

I'm also working on my review skills so please give me feedback on if I'm being to picky or if I'm misunderstanding the intent behind any of your code.

File: server/index.js [hunk 2]
```diff
@@ -69,13 +72,13 @@ app.get('/products', (req, res) => {
         product.related[relatedProduct.id].info = relatedProduct;
         product.related[relatedProduct.id].styles = relatedStyles;
       }
-      res.send(200, product);
+      // res.send(200, product);
+      res.status(200).send(product)
```
[the remark this answers is not in the document]
Comment (DevonL1010) [reply server/index.js:76]: vscode marks it as  error and suggests rewriting in this way.

File: server/index.js [hunk 3]
```diff
@@ -37,6 +38,8 @@ app.get('/products', (req, res) => {
     product.cart = result[6].data;
     product.answers = {};
     product.related = {};
+    // console.log(product)
```
[the remark this answers is not in the document]
Comment (DevonL1010) [reply server/index.js:41]: sure.

File: src/components/related/OutfitList.jsx [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+
+  useEffect(() => {
+    // get the overview product info
+    axios.get(`${API_URL}/40365`,options)
+    .then(response => setOverviewProduct(response.data))
+
+  }, []);
   return (
     <div className={styles.OutfitList}>
       <i class="fas fa-plus">Add to Outfit</i>
```
[the remark this answers is not in the document]
Comment (DevonL1010) [reply src/components/related/OutfitList.jsx:24]: sure

File: server/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (merosenlund) [reply server/index.js:76]: Sounds good! We can just delete the commented out way then. Thanks for fixing this!

Do you have eslint set up? I need to get that working still.

File: src/components/related/ProductCard.jsx [hunk 5]
```diff
@@ -21,43 +25,40 @@ const ProductCard = ({overviewProduct, relatedID}) =>{
...
+    // // get related product info
+    // axios.get(`${API_URL}/${relatedID}`, options)
+    // .then(results => {
+    //     // console.log(results.data)
+    //     setRelatedProduct(results.data)
+    // })
 
     //get related product image
-    axios.get(`${API_URL}/${relatedID}/styles`, options)
+    axios.get(`${API_URL}/${relatedProduct.info.id}/styles`, options)
```
[the remark this answers is not in the document]
Comment (DevonL1010) [reply src/components/related/ProductCard.jsx]: I thought the image data was not available. I will update the code. Thanks!

File: src/components/related/ProductCardList.jsx (hunk 1 above)
Comment (DevonL1010) [reply src/components/related/ProductCardList.jsx]: The comment-out code is kept for possible future optimization. I think the way we grouping all the data together and then send to the client is not really efficient. I will move all the comment-out code to the very button.

Comment (merosenlund) [reply src/components/related/ProductCardList.jsx]: You are right there is definitely lots of room to improve the server code and optimize how we want to get the data we need.

**Comment by DevonL1010** [comment]
- moved all the comment-out code to the very button 
- change the 'class' to 'className'
- moved all the axios request and using useContext instead

**Review by merosenlund — APPROVED** [review]
Looks great! Thanks for making the changes, sorry if I was being too picky.

**Comment by DevonL1010** [comment]
That is a TRUE code review! Appreciate all your helps.
